### ISSUE TITLE
Using @deck.gl/arcgis w/ DeckRenderer and tilt, ScatterPlotLayer doesn't align with map

### ISSUE THREAD

**Issue by pdnellius** [visgl/deck.gl#5039]
#### Description
When I use a ScatterPlotLayer with a DeckRenderer, I'm seeing significant drift of the ScatterPlotLayer from the map. Attaching screenshot

#### Repro Steps
- Create ScatterPlotLayer with points that cover the US
- Give the DeckRenderer tilt
- Pan to the left or right when viewing the United States, as the perspective shifts, you will see the layer misalign/drift

![image](https://user-images.githubusercontent.com/5667111/96310507-85130900-0fd5-11eb-9063-fac50a5191f2.png)

#### Environment (please complete the following information):
 - Framework Version: deck.gl 8.2.10
 - Browser Version: Chrome 86
 - OS: macOS 10.15.7

**Comment by pdnellius** [comment]
I can see this easily when I use the following
```
sceneView.goTo({ center: coordinate, zoom: 5, tilt: 40 })
```

**Comment by Pessimistress** [comment]
@damix911 I do not believe DeckRenderer currently matches the fov between ArcGIS and deck. Deck's default fov is `Math.atan(1 / 3) * 2` ~= 36 degrees.

**Comment by pdnellius** [comment]
@damix911 I tried setting fov via `SceneView`'s `camera` object, but this did not seem to set correctly. Is there something that's overriding this setting in the `@deck.gl/arcgis` submodule source?

**Comment by Pessimistress** [comment]
There's no definition of `fov` in ArcGIS' documentation. After some experiments, I believe its fov is the angle between opposing corners of the viewport. While deck.gl's fov is `fovy`.

To match deck.gl's constant `fovy`, the base map must change its fov based on the viewport aspect ratio (`width / height`):

```js
function getFOV(aspectRatio) {
  const D = Math.sqrt(1 + aspectRatio ** 2);
  const halfFOV = Math.atan(D / 2 / 1.5); // 1.5x height is deck.gl's default camera altitude
  return halfFOV * 2 / Math.PI * 180;
}
```

I tried setting SceneView's `camera.fov` dynamically on `resize`, but it doesn't seem to work. The expected behavior should be `DeckRenderer` changing deck.gl's `viewState.altitude` to match SceneView's fov:

```js
function arcgisFOVToDeckAltitude(fov, aspectRatio) {
  const D = Math.sqrt(1 + aspectRatio ** 2);
  const halfFOV = fov / 2 / 180 * Math.PI;
  return D / 2 / Math.tan(halfFOV);
}
```

**Comment by pdnellius** [comment]
@Pessimistress I thought that this was the documentation for arcGIS fov: https://developers.arcgis.com/javascript/latest/api-reference/esri-Camera.html#fov

However I tried cloning the camera and applying this and it didn't seem to stick. Not sure if @damix911 has any other suggestions.

**Comment by Pessimistress** [comment]
> I thought that this was the documentation for arcGIS fov

Yes, but they don't provide a definition for said fov. It can refer to a number of different things, which happens to be different from what deck.gl (or most WebGL libraries) uses.

> I tried cloning the camera and applying this and it didn't seem to stick

If you set `camera.fov` on initial construction it will work. Not sure how to handle map resize, though.

As I said in the previous comment, this should really be handled in the `@deck.gl/arcgis` module, not the application itself. If you really want it to work with the current release you can create a Codesandbox and we'll collaborate from there.

**Comment by damix911** [comment]
I can confirm that the fov is corner to corner. I have a branch in which I account for that and do the conversion explicitly but it's still not enough to fix it in all cases.

**Comment by damix911** [comment]
I did a few more experiments [here](https://github.com/visgl/deck.gl/compare/master...damix911:damix911/fix3darcgis#diff-41609f296840d14f11ffdd68419c288c2aff890f24d41c68583536963db8c290). I tried to use `FirstPersonView` and properly set `fovy` and `position` to match ArcGIS.

Today I am going to try what @Pessimistress suggested which is (correct me if I wrong) try to change ArcGIS camera to match deck.gl.
